providers/test/Test.fst

```diff
@@ -296,31 +296,46 @@ let rec test_aead_loop (LB len vs) =
 let test_aead () : St unit =
   test_aead_loop aead_vectors_low
 
-let rec test_chacha20poly1305_loop (i: U32.t): St unit =
-  let open Test.Vectors.Chacha20Poly1305 in
-  if i `U32.gte` vectors_len then
-    ()
+#push-options "--z3rlimit 32"
+
+let test_chacha20poly1305_one (v: Test.Vectors.Chacha20Poly1305.vector): Stack unit (fun _ -> True) (fun _ _ _ -> True) =
```

Comment:
can this one go in Test.NoHeap since it is now in `Stack`?

Reply:
OK, will do. Same for poly1305 and curve25519 above. Then I let you decide whether I should also add it to "the Wasm tests" (`Test.NoHeap.main`)

Reply:
Yes please